### PR TITLE
Convert vld1_lane_* and other methods to const generics

### PR DESCRIPTION
**Pull request by SparrowLii**
Use generic const in the following methods:
vld1_lane_*
vld1q_lane_*
vget_lane_*
vgetq_lane_*
vshrq_n_*
vshlq_n_*
vextq_*

### CODE REVIEW

File: crates/core_arch/src/arm/neon/mod.rs [hunk 1]
```diff
@@ -589,352 +589,264 @@ pub unsafe fn vld1q_f32(ptr: *const f32) -> float32x4_t {
...
-        0 <= lane && lane <= 1,
-        "must have 0 ≤ lane ≤ 1, but lane = {}",
-        lane
-    );
-    simd_insert(src, lane as u32, *ptr)
+#[rustc_legacy_const_generics(2)]
+#[cfg_attr(all(test, target_arch = "arm"), assert_instr("vldr", LANE = 1))]
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(ld1, LANE = 1))]
+pub unsafe fn vld1q_lane_s64<const LANE: i32>(ptr: *const i64, src: int64x2_t) -> int64x2_t {
+    static_assert!(LANE : i32 where 0 <= LANE && LANE <= 1);
```
Comment (Amanieu) [inline crates/core_arch/src/arm/neon/mod.rs:680]: Add `static_assert_imm1`.

File: crates/core_arch/src/arm/neon/mod.rs [hunk 2]
```diff
@@ -589,352 +589,264 @@ pub unsafe fn vld1q_f32(ptr: *const f32) -> float32x4_t {
...
-        0 <= lane && lane <= 0,
-        "must have 0 ≤ lane ≤ 0, but lane = {}",
-        lane
-    );
-    simd_insert(src, lane as u32, *ptr)
+#[rustc_legacy_const_generics(2)]
+#[cfg_attr(all(test, target_arch = "arm"), assert_instr("vldr", LANE = 0))]
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(ldr, LANE = 0))]
+pub unsafe fn vld1_lane_s64<const LANE: i32>(ptr: *const i64, src: int64x1_t) -> int64x1_t {
+    static_assert!(LANE : i32 where 0 <= LANE && LANE <= 0);
```
Comment (Amanieu) [inline crates/core_arch/src/arm/neon/mod.rs:668]:
```suggestion
    static_assert!(LANE : i32 where LANE == 0);
```

Comment (SparrowLii) [reply crates/core_arch/src/arm/neon/mod.rs]: Thanks, has been corrected

**Comment by tesuji** [comment]
@SparrowLii 
CI broke with this PR: https://github.com/rust-lang/stdarch/pull/1028/checks?check_run_id=2003643764#step:14:486

```rust
    --> crates/core_arch/src/arm/neon/mod.rs:2985:36
     |
2985 | pub unsafe fn vgetq_lane_u64<const imm5: i32>(v: uint64x2_t) -> u64 {
     |                                    ^^^^ help: convert the identifier to upper case: `IMM5`
     |
     = note: `-D non-upper-case-globals` implied by `-D warnings`

```

**Comment by SparrowLii** [comment]
@tesuji Thanks for pointing out. I currently cannot test armv7 and arm by myself. I will make the corresponding changes

**Comment by tesuji** [comment]
No worry! I'll fix them in  #1036 .

**Comment by SparrowLii** [comment]
Thanks!

**Pull request merged**